dali/server/daldap.cpp

```diff
@@ -164,23 +166,36 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
 
                 CDateTime now;
                 now.setNow();
+                if (requestSignatureAllowedClockVarianceSeconds)//allow for clock variance between machines
+                {
+                    unsigned int hour, min, sec, nanosec;
+                    now.getTime(hour, min, sec, nanosec);
+                    now.setTime(hour, min, sec + requestSignatureAllowedClockVarianceSeconds, nanosec);
+                }
+
                 if (now.compare(reqUTCTimestamp, false) < 0)//timestamp from the future?
                 {
                     StringBuffer localDaliTimeUTC;
                     now.getString(localDaliTimeUTC, false);//get UTC timestamp
-                    ERRLOG("LDAP: getPermissions(%s) scope=%s user=%s Request digital signature UTC timestamp %s from the future (Dali UTC time %s)",key?key:"NULL",obj?obj:"NULL",username.str(), requestTimestamp.str(), localDaliTimeUTC.str());
+                    ERRLOG("LDAP: getPermissions(%s) scope=%s user=%s Request digital signature UTC timestamp %s from the future (Dali adjusted UTC time %s). Check configured allowedClockVariance (%d sec)",key?key:"NULL",obj?obj:"NULL",username.str(), requestTimestamp.str(), localDaliTimeUTC.str(), requestSignatureAllowedClockVarianceSeconds);
                     return SecAccess_None;//deny
                 }
 
                 CDateTime expiry;
                 expiry.set(now);
                 expiry.adjustTime(requestSignatureExpiryMinutes);//compute expiration timestamp
+                if (requestSignatureAllowedClockVarianceSeconds)//allow for clock variance between machines
```

Comment:
Rather than reversing the variance here, might be cleaner/more intuitive to move up the expiry initialization to before the first requestSignatureAllowedClockVarianceSeconds adjustment.  

Let everything else follow:

CDateTime now;
now.setNow();
CDateTime expiry;
expiry.set(now);
expiry.adjustTime(requestSignatureExpiryMinutes); //compute expiration timestamp
...
everything else, with no second variance adjustment.

Reply:
Agreed, good catch

Reply:
Ok, the last fix makes it clear you are not just "adjusting back", rather you are using requestSignatureAllowedClockVarianceSeconds to create a window in each direction.